File: GarbageV5/app/src/main/java/dk/itu/garbagev5/ListFragment.java
package dk.itu.garbagev5;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class ListFragment extends Fragment {


    private static ItemsDB itemsDB;
    private TextView listThings;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        itemsDB= ItemsDB.get();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        final View v = inflater.inflate(R.layout.fragment_list, container, false);
        listThings = v.findViewById(R.id.listItems);
        listThings.setText(" Sorting List" + itemsDB.listItems());
        return v;
    }
        // Inflate the layout for this fragment
}

File: GarbageV6/app/src/main/java/dk/itu/garbagev6/ListFragment.kt
package dk.itu.garbagev6

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider

class ListFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val v = inflater.inflate(R.layout.fragment_list, container, false)
        val listThings:TextView = v.findViewById(R.id.listItems)

        val viewModel = ViewModelProvider(requireActivity())[MainViewModel::class.java]

        viewModel.uiState.observe(viewLifecycleOwner) { uiState ->
            listThings.text = "Sorting List\n" + uiState.listItems
        }

        return v
    }
}